using/self_service.md

```diff
@@ -322,3 +322,118 @@ Once allocation has been created, you can see its access information:
+
+Now, navigate to Resources -> Applications on the left. Then press Create application in the top right corner of the table and select Python management:
+
+![Available applications popup](../images/available-applications-list.png)
+
+Before saving the form, follow the instructions under “Virtual machine has cloud broker public key” checkbox:
+
+1. Find out External IP of your VM
+   ![VM External IP](../images/vm-external-ip.png)
+
+2. Connect with PuTTY (private key should correspond to the public key you uploaded to ETAIS!) to your VM
+   ![Connect via Putty](../images/connect-putty.png)
+
+3. Or use OpenSSH on Linux/MacOS: ssh TARGET_USER@VM_EXTERNAL_IP
+
+Once you are logged in (default system user depends on a distribution: either debian or ubuntu), copy cloud broker public key from the Python management form and paste it into /home/YOUR_TARGET_SYSTEM_USER/.ssh/authorized_keys
+
+![Pasting ETAIS public key into authorized_keys on your VM](../images/python-management-authorized-keys.png)
+
+Now you can proceed with Python management form and afterwards cotinue with [JupyterHub management](#jupyterhub-management-service).
+
+### Details screen elements
+
+![Python management details screen elements](../images/python-management-details.png)
+
+> Virtual environments are managed using virtualenv and virtualenvwrapper utilities.
+
+Python Management service is not notified when you install additional libraries manually.
+
+> If you install any libraries directly or create new virtual environments via SSH, click either Find missing libraries or Find installed virtual environments & libraries.
+
+1. Button to propagate changes to a virtual machine (install/delete libraries, create/delete virtual environments)
+2. Virtual environment name (a folder inside Virtual environments root directory)
+3. Download installed libraries list in a form of [requirements file format](https://pip.pypa.io/en/stable/user_guide/#requirements-files). Currently supports only exact versions of packages: pkg==1.1
+4. Upload requirements.txt for installation in a virtual environment. Again, only exact versions of packages. New libraries will appear in the form, save to apply the changes.
+5. If any libraries were installed manually using pip through SSH (or via Jupyter notebooks), you need to synchronize that virtual environment with ETAIS using this button (otherwise new libraries will not appear in a virtual environmnet)
+6. If any virtual environments were manually installed/removed in Virtual environments root directory, you need to synchronize the state with ETAIS with this button (otherwise new virtual environment will not appear).
+7. A brief guide which tells you how to activate virtual environment in terminal, install libraries and that you should click “Find missing installed libraries” button if new libraries were installed manually.
+8. Each action performed by ETAIS on a VM is logged here.
```

Comment:
Where? Why is it important?

Reply:
Here!

![image](https://user-images.githubusercontent.com/8850358/38744949-de82382e-3f4b-11e8-8235-98cc09811cf8.png)